// @hyvor/laravel-playwright@2.0.2/dist/laravel-playwright.js
import { test as c } from "@playwright/test";
const o = c.extend({
  laravelBaseUrl: [void 0, { option: !0 }],
  laravel: async ({ laravelBaseUrl: s, baseURL: t, request: a }, r) => {
    const n = s || t + "/playwright", e = new l(n, a);
    await r(e), await e.tearDown();
  }
});
class l {
  constructor(t, a) {
    this.baseUrl = t, this.request = a;
  }
  async call(t, a = {}) {
    const r = this.baseUrl.replace(/\/$/, "") + t, n = await this.request.post(r, { data: a });
    if (n.status() !== 200)
      throw new Error(`
                Failed to call Laravel ${t}. 
                Status: ${n.status()}
                Response: ${await n.text()}
            `);
    return await n.json();
  }
  async artisan(t, a = []) {
    return await this.call("/artisan", { command: t, parameters: a });
  }
  async truncate(t = []) {
    return await this.call("/truncate", { connections: t });
  }
  async factory(t, a = {}, r = void 0) {
    return await this.call("/factory", { model: t, count: r, attrs: a });
  }
  async query(t, a = [], r = {}) {
    const { connection: n = null, unprepared: e = !1 } = r;
    if (e && a.length > 0)
      throw new Error("Cannot use unprepared with bindings");
    return await this.call("/query", {
      query: t,
      bindings: a,
      connection: n,
      unprepared: e
    });
  }
  async select(t, a = {}, r = {}) {
    const { connection: n = null } = r;
    return await this.call("/select", { query: t, bindings: a, connection: n });
  }
  async callFunction(t, a = []) {
    return await this.call("/function", { function: t, args: a });
  }
  /**
   * Sets a laravel config value until tearDown is called (or the test ends)
   */
  async config(t, a) {
    return await this.call("/dynamicConfig", { key: t, value: a });
  }
  /**
   * Travel to a specific time
   * ex: travel('2021-01-01 00:00:00')
   */
  async travel(t) {
    return await this.call("/travel", { to: t });
  }
  async registerBootFunction(t) {
    return await this.call("/registerBootFunction", { function: t });
  }
  async tearDown() {
    return await this.call("/tearDown");
  }
}
export {
  l as Laravel,
  o as test
};
